api/controllers/v0.2/createPatron.js

```diff
@@ -135,25 +216,42 @@ function createPatron(req, res) {
         collectErrorResponseData(
           null,
           'invalid-request',
-          'Missing required patron information.',
+          'Required patron information is missing.',
           null,
           debugMessage // eslint-disable-line comma-dangle
         ) // eslint-disable-line comma-dangle
       ));
-
-    return;
+    return false;
   }
+  return true;
+}
+
+/**
+ * createPatron(req, res)
+ * The callback for the route "/patrons".
+ * It will fire a POST request to the ILS API for creating a new patron.
+ *
+ * @param {HTTP request} req
+ * @param {HTTP response} res
+ */
+function createPatron(req, res) {
+  if (validateEnvironmentAndRequest(req, res) === false) { return false; }
 
   ilsClientKey = ilsClientKey || awsDecrypt.decryptKMS(process.env.ILS_CLIENT_KEY);
   ilsClientPassword = ilsClientPassword || awsDecrypt.decryptKMS(process.env.ILS_CLIENT_SECRET);
 
+  // eslint-disable-next-line max-len
+  // Delete pcode4 data to prevent an error on Sierra TODO: remove this line once pcode4 is accepted; also remove related line in v0.2 modelStreamPatron.js
+  if (req.body && req.body.patronCodes) {
+    delete req.body.patronCodes.pcode4;
```

Comment:
Just want to make sure I understand this. So sierra patron api fails if you send it a pcode4, so you're removing pcode4 from the posted data and plan to submit another PR in the future to remove these places where you're removing `pcode4`?

Reply:
Yes, there will be a new PR.  Sierra fails if pcode4 is included and the stream fails if pcode4 is not included.  Publishing to the stream happens afterwards so the sequence is to remove and then add.